specification/cost-management/resource-manager/Microsoft.CostManagement/stable/2022-10-01/costmanagement.generatedetailedcostreport.json

```diff
@@ -0,0 +1,373 @@
+            "schema": {
+              "$ref": "#/definitions/GenerateDetailedCostReportOperationStatuses"
+            }
+          },
+          "default": {
+            "description": "Error response describing why the operation failed.",
+            "schema": {
+              "$ref": "common-types.json#/definitions/ErrorResponse"
+            }
+          }
+        }
+      }
+    }
+  },
+  "definitions": {
+    "GenerateDetailedCostReportErrorResponse": {
+      "description": "Error response indicates that the service is not able to process the incoming request. The reason is provided in the error message. \n\nSome Error responses: \n\n * 413 Request Entity Too Large - Request is throttled. The amount of data required to fulfill the request exceeds the maximum size permitted of 2Gb. Please utilize our Exports feature instead. \n\n * 429 TooManyRequests - Request is throttled. Retry after waiting for the time specified in the \"x-ms-ratelimit-microsoft.consumption-retry-after\" header. \n\n * 503 ServiceUnavailable - Service is temporarily unavailable. Retry after waiting for the time specified in the \"Retry-After\" header.",
+      "type": "object",
+      "properties": {
+        "error": {
+          "description": "The details of the error.",
+          "$ref": "common-types.json#/definitions/ErrorDetails"
+        }
+      }
+    },
+    "GenerateDetailedCostReportDefinition": {
+      "description": "The definition of a cost detailed report.",
+      "properties": {
+        "metric": {
+          "description": "The type of the detailed report. By default ActualCost is provided",
+          "type": "string",
+          "enum": [
+            "ActualCost",
+            "AmortizedCost"
+          ],
+          "x-ms-enum": {
+            "name": "GenerateDetailedCostReportMetricType",
+            "modelAsString": true
+          }
+        },
+        "timePeriod": {
+          "description": "Has time period for pulling data for the cost detailed report. Can only have one of either timePeriod or invoiceId or billingPeriod parameters. If none provided current month cost is provided.",
+          "$ref": "#/definitions/GenerateDetailedCostReportTimePeriod"
+        },
+        "billingPeriod": {
+          "description": "Billing period in YearMonth(e.g. 202008) format. Only for legacy enterprise customers can use this. Can only have one of either timePeriod or invoiceId or billingPeriod parameters. If none provided current month cost is provided.",
+          "type": "string"
+        },
+        "invoiceId": {
+          "description": "Invoice ID for PayAsYouGo customers and billing profile scope. Can only have one of either timePeriod or invoiceId or billingPeriod parameters. If none provided current month cost is provided.",
+          "type": "string"
+        },
+        "customerId": {
+          "description": "Customer ID for Modern (Invoice Id and billing profile is also required for this).",
```

Comment:
>Modern [](http://example.com/codeflow?start=42&length=6)

I thought you're dropping the term 'Modern' from documentation?

Reply:
I wasn't going to but I am now :)